pkg/operations/kubernetesupgrade/upgradecluster.go

```diff
@@ -84,21 +85,26 @@ func (uc *UpgradeCluster) UpgradeCluster(az armhelpers.AKSEngineClient, kubeConf
 		return uc.Translator.Errorf("Error while querying ARM for resources: %+v", err)
 	}
 
-	var upgrader UpgradeWorkFlow
 	upgradeVersion := uc.DataModel.Properties.OrchestratorProfile.OrchestratorVersion
 	uc.Logger.Infof("Upgrading to Kubernetes version %s\n", upgradeVersion)
-	u := &Upgrader{}
-	u.Init(uc.Translator, uc.Logger, uc.ClusterTopology, uc.Client, kubeConfig, uc.StepTimeout, aksEngineVersion)
-	upgrader = u
 
-	if err := upgrader.RunUpgrade(); err != nil {
+	if err := uc.getUpgradeWorkflow(kubeConfig, aksEngineVersion).RunUpgrade(); err != nil {
 		return err
 	}
 
 	uc.Logger.Infof("Cluster upgraded successfully to Kubernetes version %s\n", upgradeVersion)
 	return nil
 }
 
```

Comment:
getUpgradeWorkflow encapsulates the default initialization of the UpgradeWorkflow.